packages/flutter/lib/src/material/colors.dart

```diff
@@ -2,7 +2,76 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:ui' show Color;
+import 'dart:ui' show Color, hashValues;
+
+class _ColorSwatch extends Color {
+  const _ColorSwatch(int primary, this.swatch) : super(primary);
+
+  /// The shades of this color defined by material design.
+  final Map<int, Color> swatch;
```

Comment:
Mostly they could although it's a more disruptive change. The most difficult case I ran into was FlutterLogoDecoration, since it really doesn't want to depend on material.dart